### PR TITLE
Add support for revive as lintTool

### PR DESCRIPTION
**Pull request by mgechev**
Introduces support for revive. Revive is an extensible, configurable, and faster alternative of golint.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +98,14 @@ You can configure golangci-lint with `go.lintFlags`, for example to show issues
   "go.lintFlags": ["--enable-all", "--new"],
 ```
 
+An alternative of golint is [revive](https://github.com/mgechev/revive). It extensible, configurable, provides superset of the rules of golint rules, and has significantly better performance.
+
+To configure revive, use:
+
+```javascript
+  "go.lintFlags": ["-config=[PATH_TO_CONFIG.toml]"]
```
Comment (ramya-rao-a) [inline README.md]: Users might want to use the `${workspaceFolder}` or `${workspaceRoot}` variables for the config file. We support this for the config file used by `gometalinter`. You might want to do something similar for revive.

See https://github.com/Microsoft/vscode-go/blob/master/src/goLint.ts#L67

**Comment by wolfeidau** [comment]
@mgechev are you going to fix this feedback so we can try it out?

Keen to use this!

Cheers

File: README.md (hunk 1 above)
Comment (mgechev) [reply README.md]: Just updated the readme & tested the `${workspaceRoot}`/`${workspaceFolder}` support locally. Everything seems to be functioning correctly.

**Comment by ramya-rao-a** [comment]
Thanks for the PR!

I'll update this thread soon with steps to try this out before I release the next update.

**Comment by mgechev** [comment]
Sounds good! I'm already using it in my project:
 
cyclomatic

Would definitely verify again once you share the steps.

**Comment by ramya-rao-a** [comment]
To test this,

- Download https://github.com/Microsoft/vscode-go/blob/master/Go-latest.vsix
- Run `code --install-extension Go-latest.vsix` 
- If the above fails with `Error: end of central directory record signature not found`, then clone this repo and use the `Go-latest.vsix` file from the cloned repo
- Reload VS Code
- Change the `go.lintTool` setting to `revive`

**Comment by mgechev** [comment]
@ramya-rao-a verified that everything works as expected. Tested the config file as well 💯

**Pull request merged**